src/Extensions/Hosting.Services.Web/HostBuilderExtensions.cs

```diff
@@ -3,75 +3,129 @@
 
 		/// <summary>
-		/// Add Kestrel service listener to stateful service
+		/// Build stateless service with Kestrel service listeners
 		/// </summary>
+		/// <param name="hostBuilder">Host builder</param>
+		/// <param name="serviceName">Name of Service Fabric stateless service</param>
+		/// <param name="endpoints">Name of the web listener</param>
+		/// <param name="integrationOptions">Service Fabric integration options, default value is UseReverseProxyIntegration</param>
+		/// <param name="kestrelOptions">Action to configure Kestrel additional, for example witch to https</param>
+		/// <param name="serviceBuilder">Action to configure SF stateless service additional, for example to add more listeners or actions</param>
 		/// <typeparam name="TStartup">The type containing the startup methods for the web listener</typeparam>
-		public static ServiceFabricHostBuilder<OmexStatefulService, StatefulServiceContext> AddKestrelListener<TStartup>(
-			this ServiceFabricHostBuilder<OmexStatefulService, StatefulServiceContext> builder,
-			string name,
-			ServiceFabricIntegrationOptions options,
-			Action<IWebHostBuilder>? builderExtension = null,
-			Action<WebHostBuilderContext, KestrelServerOptions>? kestrelOptions = null)
-				where TStartup : class =>
-					builder.AddKestrelListener<TStartup, OmexStatefulService, StatefulServiceContext>(
-						name,
-						options,
-						builderExtension,
-						kestrelOptions);
-
-		private static ServiceFabricHostBuilder<TService, TContext> AddKestrelListener<TStartup, TService, TContext>(
-			this ServiceFabricHostBuilder<TService, TContext> builder,
-			string name,
-			ServiceFabricIntegrationOptions options,
-			Action<IWebHostBuilder>? builderExtension = null,
-			Action<WebHostBuilderContext, KestrelServerOptions>? kestrelOptions = null)
+		public static IHost BuildStatelessWebService<TStartup>(
+			this IHostBuilder hostBuilder,
+			string serviceName,
+			WebEndpointInfo[] endpoints,
+			ServiceFabricIntegrationOptions integrationOptions = DefaultServiceFabricIntegrationOptions,
+			Action<WebHostBuilderContext, KestrelServerOptions>? kestrelOptions = null,
+			Action<ServiceFabricHostBuilder<OmexStatelessService, StatelessServiceContext>>? serviceBuilder = null)
 				where TStartup : class
-				where TService : IServiceFabricService<TContext>
-				where TContext : ServiceContext =>
-					builder.AddServiceListener(provider => new KestrelListenerBuilder<TStartup, TService, TContext>(
-						name,
-						provider,
-						options,
-						builder => builder.BuilderExtension<TContext>(builderExtension),
-						kestrelOptions ??= s_emptyKestrelOptions));
+		{
+			if (integrationOptions.HasFlag(ServiceFabricIntegrationOptions.UseUniqueServiceUrl))
+			{
+				// Supporting this options would require some hacks:
+				// * Creating UrlSuffix like PartitionId/InstanceId where PartitionId could be taken from Env variable 'Fabric_PartitionId', for InstanceId we might thy to use new guid instead
```

Comment:
are we going to add support later? If yes, can you please create an issue?

Reply:
Currently, we are not using it and hope it wouldn't be needed, I've added a comment after my investigation, in case we would decide to do it in the future